client/state/connectionMiddleware/middleware.js

```diff
@@ -0,0 +1,19 @@
+/**
+ * Internal dependencies
+ */
+import wpcom from 'lib/wp';
+
+const connections = {};
+
+export const connectionMiddleware = store => next => action => {
+	if ( connections.hasOwnProperty( action.type ) ) {
+		connections[ action.type ].connectionFunction( wpcom )( action, store.dispatch );
+		next( action );
+	} else {
+		next( action );
+	}
+};
+
+export function registerConnection( actionType, connectionFunction, options = {} ) {
```

Comment:
@artpi Have you tested it with hot reloading? It shouldn't be a big deal when you update `actions.js` file, because it should register connection again and override old assignment. However when you change `actionType` or even remove one, the connection middleware can start to behave strangely.